qemu/tests/live_snapshot_basic.py

```diff
@@ -95,25 +89,9 @@ def check_snapshot(self):
                                           % (self.snapshot_node_name,
                                              snapshot_info))
 
-
-@error_context.context_aware
-def run(test, params, env):
-    vm = env.get_vm(params["main_vm"])
-
-    image_tag = params.get("image_name", "image1")
-    image_params = params.object_params(image_tag)
-    snapshot_test = LiveSnapshotBase(image_params, env)
-
-    error_context.context("Creating snapshot", logging.info)
-    snapshot_test.create_snapshot()
-    error_context.context("Checking snapshot created successfully",
-                          logging.info)
-    snapshot_test.check_snapshot()
-
-    sub_type = params.get("sub_type_after_snapshot")
-    if sub_type:
-        error_context.context("%s after snapshot" % sub_type, logging.info)
-        utils_test.run_virt_sub_test(test, params, env, sub_type)
-
-    if vm.is_alive():
-        vm.destroy()
+    def action_after_finished(self):
+        """
+        Run steps after live snapshot done.
+        """
+        params = self.parser_test_args()
```

Comment:
Used params here in parser_test_args().
params = self.params.object_params(self.tag)